hydra-cluster/bench/Bench/EndToEnd.hs

```diff
@@ -95,57 +96,116 @@ bench startingNodeId timeoutSeconds workDir dataset@Dataset{clientDatasets, titl
-
-              deadline <- waitMatch 300 leader $ \v -> do
-                guard $ v ^? key "tag" == Just "HeadIsClosed"
-                guard $ v ^? key "headId" == Just (toJSON headId)
-                v ^? key "contestationDeadline" . _JSON
-
-              -- Expect to see ReadyToFanout within 3 seconds after deadline
-              remainingTime <- diffUTCTime deadline <$> getCurrentTime
-              waitFor hydraTracer (remainingTime + 3) [leader] $
-                output "ReadyToFanout" ["headId" .= headId]
-
-              putTextLn "Finalizing the Head"
-              send leader $ input "Fanout" []
-              waitMatch 100 leader $ \v -> do
-                guard (v ^? key "tag" == Just "HeadIsFinalized")
-                guard $ v ^? key "headId" == Just (toJSON headId)
-
-              let res = mapMaybe analyze . Map.toList $ processedTransactions
-                  aggregates = movingAverage res
-
-              writeResultsCsv (workDir </> "results.csv") aggregates
-
-              let confTimes = map (\(_, _, a) -> a) res
-                  numberOfTxs = length confTimes
-                  numberOfInvalidTxs = length $ Map.filter (isJust . invalidAt) processedTransactions
-                  averageConfirmationTime = sum confTimes / fromIntegral numberOfTxs
-                  quantiles = makeQuantiles confTimes
-                  summaryTitle = fromMaybe "Baseline Scenario" title
-                  summaryDescription = fromMaybe defaultDescription description
-
-              pure $ Summary{clusterSize, numberOfTxs, averageConfirmationTime, quantiles, summaryTitle, summaryDescription, numberOfInvalidTxs}
+              scenario hydraTracer node workDir dataset parties leader followers
+
+benchDemo ::
+  NetworkId ->
+  SocketPath ->
+  NominalDiffTime ->
+  VerificationKey PaymentKey ->
+  [SigningKey HydraKey] ->
+  FilePath ->
+  Dataset ->
+  IO Summary
+benchDemo networkId nodeSocket timeoutSeconds faucetVk hydraKeys workDir dataset@Dataset{clientDatasets, fundingTransaction} = do
+  putStrLn $ "Test logs available in: " <> (workDir </> "test.log")
+  withFile (workDir </> "test.log") ReadWriteMode $ \hdl ->
+    withTracerOutputTo hdl "Test" $ \tracer ->
+      failAfter timeoutSeconds $ do
+        putTextLn "Starting benchmark"
+        let cardanoTracer = contramap FromCardanoNode tracer
+        findRunningCardanoNode' cardanoTracer networkId nodeSocket >>= \case
+          Nothing ->
+            error ("Not found running node at socket: " <> show nodeSocket <> ", and network: " <> show networkId)
+          Just node -> do
+            let clientSks = clientKeys <$> clientDatasets
+            (`finally` returnFaucetFunds tracer node clientSks) $ do
+              putTextLn "Seeding network"
+              fundClients networkId nodeSocket fundingTransaction
+              forM_ clientSks (fuelWith100Ada (contramap FromFaucet tracer) node)
+              putStrLn $ "Connecting to hydra cluster in " <> workDir
```

Comment:
The cluster is not running in a work dir. Instead, we are connecting to the three (expected to run) hydra nodes on ports 4001, 4002 and 4003 here.

Should: put that on the stdout instead